{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "557ec155-0303-4dd8-894c-0d357cfca209",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load hisat2/2.2.0\n",
    "\n",
    "hisat2-build GCF_002042975.1_ofav_dov_v1_genomic.fna GCF_002042975.1_ofav_dov_v1_genomic_index --threads 15\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6701dd52-b3b5-40d5-af40-427049b43711",
   "metadata": {},
   "outputs": [],
   "source": [
    "mv \"old_filename\" \"$(echo \"old_filename\" | sed 's/S.*_/_/')\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "32bdaba2-478c-4869-8ba2-9dc355d5209c",
   "metadata": {},
   "outputs": [],
   "source": [
    "for f in $(ls *.fq); do mv ${f} \"$(echo ${f} | sed 's/_S.*_/_/')\"; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98e440af-83f2-4be6-b26b-8cfee89e2464",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load hisat2/2.2.0\n",
    "\n",
    "for f in $(ls *.fq | sed 's/_[12].fq//g'| sort -u); do hisat2 -q -x /project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/Ofav_genome/GCF_002042975.1_ofav_dov_v1_genomic_index -1 ${f}_1.fq -2 ${f}_2.fq -S ../mapped/${f}mapofav.bam --un ../unmapped/${f}unmapofav.bam ; done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c28283a5-daaa-4c66-8bf7-43ebb18b8787",
   "metadata": {},
   "source": [
    "index the reference genome for samtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b8581d20-d06a-48af-ac7c-836b61b03cca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "samtools faidx GCF_002042975.1_ofav_dov_v1_genomic.fna\n",
    "\n",
    "\n",
    "Submitted batch job 16130712"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52ca9602-0443-45ef-ad49-336155915063",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load samtools/1.9\n",
    "\n",
    "for f in $(ls *.bam| sed 's/.bam//g'); do samtools sort ${f}.bam -l 0 -o ../sorted_bam/${f}_sorted.bam ; done\n",
    "\n",
    "\n",
    "\n",
    "Submitted batch job 16130717"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d22a9cdf-10cc-4da0-afe0-5e285068b751",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "\n",
    "angsd -b resistant.filelist -minMapQ 30 -minQ 20 -anc /project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/Ofav_genome/GCF_002042975.1_ofav_dov_v1_genomic.fna -out ../saf/Resistant -dosaf 1 -gl 1\n",
    "\n",
    "\n",
    "OFAV: Submitted batch job 16132378\n",
    "\n",
    "PSTR: Submitted batch job  16132401"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbbf8445-9db9-42ae-b44f-cac501e84be9",
   "metadata": {},
   "outputs": [],
   "source": [
    "Resistant filelist ofav/oann:\n",
    "Some issues where two of what i thought would be there arent?\n",
    "\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T1_157_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T1_35_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_97_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T2_59_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_93_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_91_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T2_71_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_87_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T3_65_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T3_127_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T3_149_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T3_161_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T4_2_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T4_16_OFAVmapofav_sorted.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fdad4fe-cc17-48a8-a134-643dde7d418a",
   "metadata": {},
   "outputs": [],
   "source": [
    "REsistant filelist pstr:\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_29_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_35_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_32_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_40_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_38_PSTRmapofav_sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e52cf12-38f3-4665-8742-4dfd79be1c4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "\n",
    "angsd -b susceptible.filelist -minMapQ 30 -minQ 20 -anc /project/pi_sarah_gignouxwolfsohn_uml_edu/Reference_genomes/Ofav_genome/GCF_002042975.1_ofav_dov_v1_genomic.fna -out ../saf/Susceptible -dosaf 1 -gl 1\n",
    "\n",
    "\n",
    "\n",
    "OFAV: \n",
    "PSTR: 16132401\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "438033a4-98fd-4ece-a59a-022f90ec8006",
   "metadata": {},
   "outputs": [],
   "source": [
    "Susceptible filelist ofav/oann\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T1_136_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T1_120_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T1_41_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_119_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_118_OANNmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_101_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/52022_BEL_CBC_T2_72_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T2_102_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T4_9_OFAVmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/OFAV/sorted_bam/122022_BEL_CBC_T4_10_OFAVmapofav_sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "afade76b-f355-4079-a97f-ae9c2437ede2",
   "metadata": {},
   "outputs": [],
   "source": [
    "Susceptible filelist of pstr:\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_25_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_27_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_26_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_31_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_28_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T1_30_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_33_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T2_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_37_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_36_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_34_PSTRmapofav_sorted.bam\n",
    "/project/pi_sarah_gignouxwolfsohn_uml_edu/SCTLD_metagenomes/PSTR/sorted_bam/102019_BEL_CBC_T3_39_PSTRmapofav_sorted.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d93cbdfb-9b5e-4539-8e57-794d38b3c28d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "\n",
    "realSFS Resistant.saf.idx Susceptible.saf.idx > ../sfs/Resistant.Susceptible.ml\n",
    "\n",
    "PSTR: Submitted batch job 16132445\n",
    "OFAV: Submitted batch job 16132447 OOM killed, trying again with 100000 mem\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "593abf60-a08e-48a2-bad7-08004788a3a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "realSFS fst index ../saf/Resistant.saf.idx ../saf/Susceptible.saf.idx -sfs Resistant.Susceptible.ml -fstout ../fst/Resistant.Susceptible\n",
    "\n",
    "PSTR: Submitted batch job 16132554\n",
    "OFAV: Submitted batch job 16150278"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2eb8d4ab-d21a-4444-8558-02b807f46446",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "\n",
    "realSFS fst print Resistant.Susceptible.fst.idx\n",
    "\n",
    "realSFS fst stats2 Resistant.Susceptible.fst.idx -win 50000 -step 10000 >Resistant.Susceptible.slidingwindow.PSTR\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dc679b0-56a7-438c-9302-af75b20522d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load angsd/0.935\n",
    "\n",
    "realSFS fst print Resistant.Susceptible.fst.idx\n",
    "\n",
    "realSFS fst stats2 Resistant.Susceptible.fst.idx -win 50000 -step 10000 >Resistant.Susceptible.slidingwindow.OFAV\n",
    "\n",
    "\n",
    "Submitted batch job 16150287\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2fef51a-0362-43fe-a340-5c5bfd33e86c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "335d91c2-b05f-4954-994c-2ee745cde450",
   "metadata": {},
   "outputs": [],
   "source": [
    "Mapping to symbiont genomes. symbiont genomes have already been indexed. created conda environment for fastq_screen. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7348e955-ccae-4d7d-910f-1657d02cfc9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH -c 4  # Number of Cores per Task\n",
    "#SBATCH -t 10-0:00\n",
    "#SBATCH --mem=60000 # Requested Memory\n",
    "#SBATCH -p cpu-preempt  # Partition\n",
    "#SBATCH -c 64  # Number of CPUs\n",
    "#SBATCH -o slurm-%j.out  # %j = job ID\n",
    "\n",
    "module load miniconda/22.11.1-1\n",
    "\n",
    "conda activate genomes\n",
    "\n",
    "\n",
    "for f in $(ls *.fq| sed 's/_[12].fq//g'); \n",
    "do fastq_screen --threads 15 --subset 0 --tag --aligner Bowtie2 \"$f\"_1.fq \"$f\"_2.fq; \n",
    "done\n",
    "\n",
    "Submitted batch job 16130938"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77789892-c0e4-43cd-9008-f989344c00e0",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
